includes/modules/management/class-wcs-att-manage-add.php

```diff
@@ -0,0 +1,497 @@
+<?php
+/**
+ * WCS_ATT_Manage_Add class
+ *
+ * @author   SomewhereWarm <info@somewherewarm.gr>
+ * @package  WooCommerce Subscribe All the Things
+ * @since    2.1.0
+ */
+
+// Exit if accessed directly.
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * Add stuff to existing subscriptions.
+ *
+ * @class    WCS_ATT_Manage_Add
+ * @version  2.1.0
+ */
+class WCS_ATT_Manage_Add extends WCS_ATT_Abstract_Module {
+
+	/**
+	 * Include sub-modules.
+	 */
+	protected function register_modules() {
+
+		// Add product to susbcription hooks.
+		require_once( 'add/class-wcs-att-manage-add-product.php' );
+		// Add cart to susbcription hooks.
+		require_once( 'add/class-wcs-att-manage-add-cart.php' );
+
+		// Initialize modules.
+		$this->modules = array(
+			'WCS_ATT_Manage_Add_Product',
+			'WCS_ATT_Manage_Add_Cart'
+		);
+	}
+
+	/**
+	 * Register hooks.
+	 *
+	 * @return void
+	 */
+	protected function register_core_hooks() {
+
+		// Adds a product to a subscription.
+		add_action( 'wcsatt_add_product_to_subscription', array( __CLASS__, 'add_product_to_subscription' ), 10, 3 );
+
+		// Adds all items in a recurring cart to a subscription.
+		add_action( 'wcsatt_add_cart_to_subscription', array( __CLASS__, 'add_cart_to_subscription' ), 10 );
+	}
+
+	/**
+	 * Get posted data.
+	 *
+	 * @param  string  $context
+	 * @return array
+	 */
+	public static function get_posted_data( $context ) {
```

Comment:
Similar to my comment on `WCS_ATT_Form_Handler::get_posted_subscription_scheme()`, because there is no overlap between the 3 contexts, it seems like this method would be better as 3 separate methods:

* `WCS_ATT_Manage_Add::get_posted_product_data()`
* `WCS_ATT_Manage_Add::get_posted_cart_data()`
* `WCS_ATT_Manage_Add::get_posted_cart_updated_data()`